HedgeTools/ForcesSCFNTGenerator/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿using HedgeLib.IO;
+using HedgeLib.Headers;
+using System.IO;
+using System;
+
+namespace ForcesSCFNTGenerator
+{
+    public class Program
+    {
+        // Variables/Constants
+        public static BINAHeader Header = new BINAHeader() { Version = 210 };
+        
+        // Methods
+        public static void Main(string[] args)
+        {
+            if (args.Length < 1)
+            {
+                Console.WriteLine("Error: No Input file was given!\n");
+                ShowHelp();
+                return;
+            }
+
+            // Checks if the input has .otf or .ttf
+            if (!CheckExtension(args[0], ".otf", ".ttf"))
+            {
+                Console.WriteLine("Error: Input file doesn't have a .otf or .ttf extension!\n");
+                ShowHelp();
+                return;
+            }
+
+            // Input
+            FileInfo fileInfo = new FileInfo(args[0]);
+            // Output
+            string outputFilePath = Path.ChangeExtension(fileInfo.FullName, ".scfnt");
+
+            // Get output path
+            if (args.Length > 1)
+                outputFilePath = args[1];
+
+            // Reads font into Memory
+            var data = File.ReadAllBytes(fileInfo.FullName);
+            // Opens a new File
+            using (var stream = File.OpenWrite(outputFilePath))
+            {
+                BINAWriter writer = new BINAWriter(stream, BINA.BINATypes.Version2, false, true);
+                writer.WriteSignature("KFCS1000");  // Signature
+                writer.Write((long)data.Length);    // FileSize
```

Comment:
I checked a couple of scfnt files, seems like only the second filesize value ("FileSize again") is actually the filesize. This value is actually an offset to the font's name within the BINA string table at the end of the file (BINA offsets are relative to the end of the header, so you have to add 0x40). It's in the offset table and everything.

If you just replace this line with:
```csharp
var name = Path.GetFileNameWithoutExtension(fileInfo.Name);
writer.AddString("FontName", name, 8);
```

That should be more accurate (and it'll handle writing the string at the end in the right spot, adding an entry to the offset table, and filling in the offset value automatically when you call writer.FinishWrite, so this is all that would need to be changed).

Reply:
Awesome, thanks once again!